injected/src/features/broker-protection/actions/expectation.js

```diff
@@ -1,23 +1,43 @@
 import { getElement } from '../utils.js'
 import { ErrorResponse, SuccessResponse } from '../types.js'
+import { execute } from '../execute.js'
 
 /**
  * @param {Record<string, any>} action
- * @param {Document | HTMLElement} root
+ * @param {Record<string, any>} userData
+ * @param {Document} root
  * @return {import('../types.js').ActionResponse}
  */
-export function expectation (action, root = document) {
+export function expectation (action, userData, root = document) {
+    /**
+     * @type {Array<import('../types.js').BooleanResult & { failSilently?: boolean }>}
+     */
     const results = expectMany(action.expectations, root)
+    const errors = []
+    let runActions = true
+
+    results.forEach((x) => {
+        if (x.result === false) {
+            runActions = false
+
+            if (!x.failSilently) {
+                errors.push('error' in x ? x.error : 'unknown error')
+            }
 
-    const errors = results.filter(x => x.result === false).map(x => {
-        if ('error' in x) return x.error
-        return 'unknown error'
+            delete x.failSilently
+        }
     })
 
     if (errors.length > 0) {
         return new ErrorResponse({ actionID: action.id, message: errors.join(', ') })
     }
 
+    if (action.actions?.length && runActions) {
+        action.actions.forEach((expectationAction) => {
+            execute(expectationAction, userData, root)
```

Comment:
execute is async, and it can fail.

I believe your intention here is just allow the more simple actions, like 'clicks' to run afterwards. Still we need to call these sequentially. you might also want to catch errors for observibility? see my commit https://github.com/duckduckgo/content-scope-scripts/commit/3676158835a8325108db37e81933d0fff462b1fa#diff-588b8450ad6aff7276f65312571cd134fa5afb0cd4fc4066d106bd74a47891f7R33-R41 here - I left a TODO to indicate where you might need to think about the errors?